.github/workflows/run_tests.yml

```diff
@@ -121,79 +112,87 @@ jobs:
+          sudo docker cp redisai:/usr/lib/redis/modules/redisai.so /usr/lib/redis/modules &&
+          sudo docker cp redisai:/usr/lib/redis/modules/backends/ /usr/lib/redis/modules/ &&
+          echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/install/lib" >> $GITHUB_ENV &&
+          echo "REDISAI_MODULES=/usr/lib/redis/modules/redisai.so \
+                TF /usr/lib/redis/modules/backends/redisai_tensorflow/redisai_tensorflow.so \
+                TORCH /usr/lib/redis/modules/backends/redisai_torch/redisai_torch.so" >> $GITHUB_ENV
+
+      - name: Single-shard DB tests
         run: |
-          redis-server --port 7000 --daemonize yes --cluster-enabled yes --cluster-config-file 7000.conf --protected-mode no --save "" --loadmodule $HOME/redisai.so TF /usr/lib/redis/modules/backends/redisai_tensorflow/redisai_tensorflow.so TORCH /usr/lib/redis/modules/backends/redisai_torch/redisai_torch.so &
-          redis-server --port 7001 --daemonize yes --cluster-enabled yes --cluster-config-file 7001.conf --protected-mode no --save "" --loadmodule $HOME/redisai.so TF /usr/lib/redis/modules/backends/redisai_tensorflow/redisai_tensorflow.so TORCH /usr/lib/redis/modules/backends/redisai_torch/redisai_torch.so &
-          redis-server --port 7002 --daemonize yes --cluster-enabled yes --cluster-config-file 7002.conf --protected-mode no --save "" --loadmodule $HOME/redisai.so TF /usr/lib/redis/modules/backends/redisai_tensorflow/redisai_tensorflow.so TORCH /usr/lib/redis/modules/backends/redisai_torch/redisai_torch.so
-
-      - name: Overwrite redis cluster env vars
+          SCRATCH_DIR="single_scratch"
+          rm -rf $SCRATCH_DIR; mkdir $SCRATCH_DIR; pushd $SCRATCH_DIR &&
+          redis-server --port 6380 --daemonize yes \
+                       --logfile "single.log" \
+                       --loadmodule $REDISAI_MODULES &&
+          sleep 5; cat single.log &&
+          popd &&
+          export SSDB=$SSDB_SINGLE SMARTREDIS_TEST_CLUSTER=False &&
+          make test-verbose COV_FLAGS="--cov=./src/python/module/smartredis/ --cov-report=xml --cov-append"
+
+      - name: Clustered DB tests
         run: |
-         echo "SSDB=127.0.0.1:7000,127.0.0.1:7001,127.0.0.1:7002" >> $GITHUB_ENV &&
-         echo "SMARTREDIS_TEST_CLUSTER=True" >> $GITHUB_ENV
-
-      - name: Start redis cluster
-        run: redis-cli --cluster create $(echo $SSDB | tr "," " ") --cluster-yes
-
-      # CI tests that should be run with clustered Redis go here
-      - name: Run testing with redis cluster
+          SCRATCH_DIR="cluster_scratch"
+          rm -rf $SCRATCH_DIR; mkdir $SCRATCH_DIR; pushd $SCRATCH_DIR &&
+          for port in {7000..7002}; do
+            redis-server --port $port --cluster-enabled yes --daemonize yes \
+                         --cluster-config-file "$port.conf" --protected-mode no --save "" \
+                         --logfile "$port.log" \
+                         --loadmodule $REDISAI_MODULES
+          done &&
+          sleep 2 &&
```

Comment:
Same concerns as above with sleeping

Reply:
Resolving this if you find the above solution acceptable